src/content/docs/en/guides/middleware.mdx

```diff
@@ -0,0 +1,176 @@
+
+```
+
+### Authentication example
+
+Here is an example of an authentication middleware:
+
+```ts
+import type { MiddlewareResponseHandler } from "astro/middleware"
+const auth: MiddlewareResponseHandler = async ({ cookies, locals }, next) => {
+	if (!cookies.has("sid")) {
+		return new Response(null, {
+			status: 405, // Not allowed
+		})
+	}
+	let sessionId = cookies.get("sid")
+	let user = await getUserFromSession(sessionId)
+	if (!user) {
+		return new Response(null, {
+			status: 405, // Not allowed
+		})
+	}
+	locals.user = user
+	return next()
+}
+export { auth as onRequest }
+```
+
+Which you can then access via your Astro components:
+
+```astro
+---
+const { user } = Astro.locals
+---
+<h1>Hello {user.name}</h1>
+```
+
+For more usage examples, please see [the middleware RFC proposal](https://github.com/withastro/roadmap/blob/rfc/stage-3-middleware/proposals/0032-middleware-api.md#middleware-workflow-and-examples)
+
+### Chaining middleware
+
+Multiple middlewares can be joined, in a specified order, using [`sequence()`](#sequence)
+
+```js title="src/middleware.js"
+import {sequence} from "astro/middleware";
+
+function validation() {}
+function auth() {}
+
+export const onRequest = sequence(validation, auth);
+```
+
+```js
+import {sequence} from "astro/middleware";
+
+async function validation(_, next) {
+    console.log("validation request");
+    const response = await next();
+    console.log("validation response");
+    return response;
+}
+async function auth(_, next) {
+  console.log("auth request");
+  const response = await next();
+  console.log("auth response");
+  return response;
+
+}
+async function greeting(_, next) {
+  console.log("greeting request");
+  const response = await next();
+  console.log("greeting response");
+  return response;
+
+}
+
+export const onRequest = sequence(validation, auth, greeting);
+```
+
+will result in the following console order:
+
+```sh
+validation request
+auth request
+greeting request
+greeting response
+auth response
+validation response
+```
+
+## API reference
+
+### onRequest()
+
+A required exported function from `src/middleware.js` that will be called before rendering every page or API route. It accepts two arguments: [context](#context) and [next()](#next)
+
+### context
+
+An object that includes information to be made available to other middleware, API routes and `.astro` routes during the rendering process. 
+
+This is a required argument passed to `onRequest()` that must contain the [`locals`](#locals) object as well as any additional properties to be shared during rendering. For example, the `context` object may include cookies used in authentication.
+
+### next ()
+
+A function that retrieves that intercepts (reads and modifies) the `Response` of a `Request` or calls the "next" middleware in the chain. For example, this function could modify the HTML body of a response.
+
+This is an optional argument of `onRequest()`.
+
+
+### locals
```

Comment:
Do you think it's worth mentioning that Astro will make these checks **only** in **dev** mode?